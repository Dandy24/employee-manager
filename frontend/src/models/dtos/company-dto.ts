export class CompanyDto {
    public name: string;
    public phone: number;
    public address: string;
    public profile_picture?: any;
}
