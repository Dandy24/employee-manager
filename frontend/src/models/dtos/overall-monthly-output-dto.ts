export class OverallMonthlyOutputDto {
    working_hours: number; //FixME django rest returns decimals as string in JSON
    sick_hours: number; //FIXME find a way to store it as a number type
    vacation_hours: number;
    start_date: string;
    end_date: string;
    // effectivity: number;
    housing_capacity: number;
}
