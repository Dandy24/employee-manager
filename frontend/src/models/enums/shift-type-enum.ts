export enum ShiftTypeEnum {
    Rano = 'ranni',
    Odpoledne = 'odpoledne',
    Vecer = 'vecer',
}
