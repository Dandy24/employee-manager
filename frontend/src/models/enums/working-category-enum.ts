export enum WorkingCategoryEnum {
    A = 'A',
    B = 'B',
    C = 'C',
}
