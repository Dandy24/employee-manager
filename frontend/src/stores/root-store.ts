import { EmployeeStore } from './employee-store';
import { CompanyStore } from './company-store';
import { action, makeObservable, observable } from 'mobx';
import { ShiftStore } from './shift-store';
import { SearchStore } from './search-store';
import { CalendarStore } from './calendar-store';
import { DashboardStore } from './dashboard-store';

import 'moment/locale/cs';
import moment from 'moment';

moment.locale('cs');

export class RootStore {
    employeeStore: EmployeeStore;
    companyStore: CompanyStore;
    shiftStore: ShiftStore;
    searchStore: SearchStore;
    calendarStore: CalendarStore;
    dashboardStore: DashboardStore;
    activePage = 'dashboard';

    constructor() {
        this.employeeStore = new EmployeeStore(this);
        this.companyStore = new CompanyStore(this);
        this.shiftStore = new ShiftStore(this);
        this.searchStore = new SearchStore(this);
        this.calendarStore = new CalendarStore(this);
        this.dashboardStore = new DashboardStore(this);

        makeObservable(this, {
            activePage: observable,

            setActivePage: action,
        });
    }

    setActivePage(newPage: string): void {
        this.activePage = newPage;
    }
}
