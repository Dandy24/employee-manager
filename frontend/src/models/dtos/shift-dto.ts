import { ShiftTypeEnum } from '../enums/shift-type-enum';

export class ShiftDto {
    date: string; //TODO date type
    time: ShiftTypeEnum;
    companyID: number; //CompanyEntity
    employeeIDs: number[]; //EmployeeEntity[]
}
