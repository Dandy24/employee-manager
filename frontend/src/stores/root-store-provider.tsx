import React from 'react';
import { RootStore } from './root-store';

const RootStoreContext = React.createContext<RootStore>({} as RootStore);

export const useRootStore = (): RootStore => {
    const context = React.useContext(RootStoreContext);
    if (context === undefined) {
        throw new Error('useStore must be used within StoreProvider');
    }

    return context;
};

interface RootStoreProviderProps {
    rootStore: RootStore;
    children: React.ReactNode;
}

export const RootStoreProvider: React.FC<RootStoreProviderProps> = ({
    rootStore,
    children,
}: RootStoreProviderProps): React.ReactElement => {
    return <RootStoreContext.Provider value={rootStore}>{children}</RootStoreContext.Provider>;
};
