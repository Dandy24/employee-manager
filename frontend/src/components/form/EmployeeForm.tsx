import React from 'react';
import { Form, Formik } from 'formik';
import { observer } from 'mobx-react-lite';
import { EmployeeDto } from '../../models/dtos/employee-dto';
import * as yup from 'yup';
import { EmployeeEntity } from '../../models/entities/employee-entity';
import { Alert, Button, Col, Row } from 'antd';
import { TextInput } from './elements/TextInput';
import { NumberInput } from './elements/NumberInput';
import { CategorySelectList } from './elements/SelectList';
import { DateSelector } from './elements/DateSelector';
import { TextArea } from './elements/TextArea';
import { CustomSwitch } from './elements/CustomSwitch';
import { CompanySelectList } from './elements/CompanySelectList';
import Dragger from 'antd/lib/upload/Dragger';
import { InboxOutlined } from '@ant-design/icons';
import { useRootStore } from '../../stores/root-store-provider';
import { ImageUpload } from './elements/image-upload';

export interface EmployeeFormProps {
    initialValues: EmployeeDto | EmployeeEntity;
    onSubmit: (values: EmployeeDto) => void;
    profile_pic: string | null;
    setProfilePic: React.Dispatch<string | ArrayBuffer>;
}

export const SUPPORTED_ATTACHMENT_FORMATS = ['application/pdf'];
export const SUPPORTED_PICTURE_FORMATS = ['image/jpeg', 'image/jpg', 'image/png'];

export const EmployeeValidationSchema = yup.object({
    first_name: yup.string().required('Pole musí být vyplněné').min(3, 'Jméno je příliš krátké'),
    last_name: yup.string().required('Pole musí být vyplněné').min(3, 'Příjmení je příliš krátké'),
    phone: yup
        .string()
        .typeError('Špatný formát čísla')
        .required('Pole musí být vyplněno')
        .matches(/^[0-9]+$/, 'Povoleny jsou pouze číslice')
        .min(12, 'Číslo musí mít přesně 12 číslic')
        .max(12, 'Číslo musí mít přesně 12 číslic'),
    email: yup.string().email('Neplatný formát emailu'),
    working_category: yup.string().required('Kategorie musí být vyplněna'),
    med_exam_date: yup.date(),
    job_assign_date: yup.date(),
    health_limitations: yup.string().max(100, 'Překročena maximální délka poznámky'),
    attachment: yup
        .mixed()
        .test('fileType', 'Podporován je pouze formát PDF', (value) =>
            !value || value?.status === 'removed' || typeof value === 'string'
                ? true
                : SUPPORTED_ATTACHMENT_FORMATS.includes(value?.type),
        ),
    profile_picture: yup
        .mixed()
        .test('fileType', 'Podporovány jsou pouze obrázky', (value) =>
            !value || value?.status === 'removed' || typeof value === 'string'
                ? true
                : SUPPORTED_PICTURE_FORMATS.includes(value?.type),
        ),
});

export const EmployeeForm: React.FC<EmployeeFormProps> = observer((props: EmployeeFormProps): JSX.Element => {
    const { initialValues, onSubmit, setProfilePic, profile_pic } = props;

    const rootStore = useRootStore();

    const { companyStore } = rootStore;

    const dummyRequest = (options) => {
        const { onSuccess } = options;
        setTimeout(() => {
            onSuccess('ok');
        }, 0);
    };

    return (
        <Formik
            initialValues={initialValues}
            onSubmit={onSubmit}
            validationSchema={EmployeeValidationSchema}
            enableReinitialize
        >
            {({ setFieldValue, errors, values, submitCount, isValid }) => (
                <Form style={{ width: '80%', marginLeft: '9%' }} data-testid="employee-form-form">
                    {!isValid && submitCount > 0 && (
                        <Alert
                            type={'error'}
                            message={'Ve formuláři jsou chyby. Opravte je a zkuste to prosím znovu.'}
                            style={{ marginBottom: '10%' }}
                            showIcon
                            data-testid={'invalid-form-error'}
                        />
                    )}

                    <Row justify="center">
                        <div data-testid={'employee-form'}>
                            <Row justify="center">
                                <Col data-testid={'profile-picture'}>
                                    <ImageUpload
                                        initialValue={initialValues?.profile_picture}
                                        value={values?.profile_picture}
                                        error={errors?.profile_picture}
                                        setFieldValue={setFieldValue}
                                        name={'profile_picture'}
                                        setProfilePic={setProfilePic}
                                        profile_pic={profile_pic}
                                    />
                                </Col>
                            </Row>

                            <div style={{ marginTop: '4.5%' }}></div>

                            <TextInput label="Jméno" name="first_name" />

                            <TextInput label="Příjmení" name="last_name" />

                            <NumberInput label="Telefon" name="phone" />

                            <TextInput label="Email" name="email" />

                            <CategorySelectList name="working_category" label="Kategorizace práce" />

                            <DateSelector name="med_exam_date" label="Datum vstupní prohlídky" />

                            <DateSelector name="job_assign_date" label="Datum nástupu do práce" />

                            <TextArea
                                name="health_limitations"
                                label="Zdravotní omezení"
                                rows={3}
                                textareaSize="large"
                            />

                            <div style={{ marginTop: '10%' }}></div>

                            {initialValues?.first_name?.length > 1 ? (
                                <CustomSwitch label="Aktivní" name="active" />
                            ) : null}

                            {initialValues?.first_name?.length > 1 && companyStore.companies ? (
                                <div style={{ marginBottom: '10%' }}>
                                    <CompanySelectList
                                        companies={companyStore.companies}
                                        name="company"
                                        label="Firma"
                                    />
                                </div>
                            ) : null}

                            <Row justify="center" data-testid={'attachments-dropzone'}>
                                {/*<FileUpload*/}
                                {/*    value={values?.attachment}*/}
                                {/*    setFieldValue={setFieldValue}*/}
                                {/*    initialValue={initialValues?.attachment}*/}
                                {/*    error={errors?.attachment}*/}
                                {/*    onClear1={(initialValues?.attachment = null)}*/}
                                {/*    onClear2={(values?.attachment = null)}*/}
                                {/*    name={'attachment'}*/}
                                {/*/>*/}
                                <Dragger
                                    customRequest={dummyRequest}
                                    onChange={(file) => setFieldValue('attachment', file.file)}
                                    name="attachment"
                                    maxCount={1}
                                    accept=".pdf"
                                    data-testid={'attachments-dropzone-input'}
                                    onRemove={() => {
                                        setFieldValue('attachment', null);
                                        initialValues.attachment = null;
                                        values.attachment = null;
                                    }}
                                    fileList={
                                        initialValues?.attachment
                                            ? [
                                                  {
                                                      uid: '',
                                                      url: initialValues?.attachment,
                                                      name: 'attachment',
                                                  },
                                              ]
                                            : values?.attachment?.name && values?.attachment?.originFileObj
                                            ? [
                                                  {
                                                      uid: '',
                                                      url: '',
                                                      name: values?.attachment?.name,
                                                  },
                                              ]
                                            : null
                                    }
                                    style={{ width: '100%' }}
                                >
                                    <p className="ant-upload-drag-icon">
                                        <InboxOutlined />
                                    </p>
                                    <p className="ant-upload-text">
                                        Klikněte do vyznačené zóny nebo přetáhněte soubor z počítače pro jeho nahrání
                                    </p>
                                    <p className="ant-upload-hint">
                                        Zde můžete nahrát například pracovní smlouvy, lékařské zprávy atd.
                                    </p>
                                    <p className="ant-upload-hint">Pouze pro dokumenty formátu PDF</p>
                                </Dragger>
                                {errors.attachment && values.attachment ? (
                                    <Alert
                                        style={{ width: '98%', marginTop: '5%' }}
                                        message={errors.attachment}
                                        type="error"
                                        showIcon
                                        data-testid="attachment-input-error"
                                    />
                                ) : null}
                            </Row>

                            <Col span={12} offset={9} style={{ marginTop: '3vh' }}>
                                <Button type="primary" htmlType="submit" data-testid="submit-button">
                                    Uložit
                                </Button>
                            </Col>
                        </div>
                    </Row>
                </Form>
            )}
        </Formik>
    );
});
