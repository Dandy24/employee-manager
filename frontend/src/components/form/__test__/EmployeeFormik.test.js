import { act, fireEvent, render, screen, waitFor } from '@testing-library/react';
import React from 'react';
import { EmployeeForm } from '../EmployeeForm';
import { WorkingCategoryEnum } from '../../../models/enums/working-category-enum';
import { RootStore } from '../../../stores/root-store';
import { RootStoreProvider } from '../../../stores/root-store-provider';
import { BrowserRouter } from 'react-router-dom';
import { EmployeeDto } from '../../../models/dtos/employee-dto';
import userEvent from '@testing-library/user-event';

jest.mock('antd', () => {
    const antd = jest.requireActual('antd');

    const Select = (props) => {
        const multiple = ['tags', 'multiple'].includes(props.mode);

        return (
            <select
                value={props.value}
                defaultValue={props.defaultValue}
                multiple={multiple}
                disabled={props.disabled}
                data-testid={props['data-testid']}
                className={props.className}
                onChange={(e) =>
                    props.onChange(
                        multiple ? Array.from(e.target.selectedOptions).map((option) => option.value) : e.target.value,
                    )
                }
            >
                {props.children}
            </select>
        );
    };

    Select.Option = ({ children, ...otherProps }) => <option {...otherProps}>{children}</option>;
    Select.OptGroup = ({ children, ...otherProps }) => <optgroup {...otherProps}>{children}</optgroup>;

    return { ...antd, Select };
});

export const initialEmployee = {
    id: 123,
    first_name: 'Test',
    last_name: 'Employee',
    email: 'test@email.com',
    phone: 420123456789,
    active: true,
    med_exam_date: '2022-02-01',
    job_assign_date: '2022-02-12',
    working_category: WorkingCategoryEnum.C,
    health_limitations: '',
    company: null,
};

// Defining missing window.matchMedia() property from JSDOM, to prevent Jest from crashing
// https://jestjs.io/docs/manual-mocks#mocking-methods-which-are-not-implemented-in-jsdom
Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // deprecated
        removeListener: jest.fn(), // deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
    })),
});

test('Form renders correctly with proper initial values', async () => {
    const handleSubmit = jest.fn();

    const { asFragment } = render(<EmployeeForm initialValues={new EmployeeDto()} onSubmit={handleSubmit} />);

    const form = await screen.findByTestId('employee-form-form');

    expect(form).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
});

test('Form display correct initial values', async () => {
    const handleSubmit = jest.fn();

    const { getAllByRole } = render(
        <RootStoreProvider rootStore={new RootStore()}>
            <BrowserRouter>
                <EmployeeForm initialValues={initialEmployee} onSubmit={handleSubmit} />
            </BrowserRouter>
        </RootStoreProvider>,
    );

    const form = await screen.findByTestId('employee-form-form');

    const firstNameInput = await screen.findByLabelText('Jméno');
    const activeCheckbox = await screen.findByRole('switch');
    const medExamInput = await screen.findByLabelText('Datum vstupní prohlídky');
    // eslint-disable-next-line testing-library/prefer-screen-queries
    const workCategoryInput = getAllByRole('combobox')[0];

    expect(firstNameInput).not.toBeNull();
    expect(firstNameInput).toHaveValue('Test');

    expect(workCategoryInput).not.toBeNull();
    expect(workCategoryInput).toHaveTextContent('B');

    expect(activeCheckbox).toBeChecked();

    expect(medExamInput).not.toBeNull();
    expect(medExamInput).toHaveValue('2022-02-01');
});

test('Form submits profile picture correctly', async () => {
    const handleSubmit = jest.fn();

    const fakeFile = [new File(['fake image data'], 'hello.png', { type: 'image/png' })];
    const fakeDoc = [new File(['fake PDF document'], 'document.pdf', { type: 'application/pdf' })];

    render(
        <RootStoreProvider rootStore={new RootStore()}>
            <BrowserRouter>
                <EmployeeForm initialValues={initialEmployee} onSubmit={handleSubmit} />
            </BrowserRouter>
        </RootStoreProvider>,
    );

    const profilePictureInput = await screen.findByTestId('profile_picture-input');
    // userEvent.upload(profilePictureInput, [fakeFile], {}, { applyAccept: true });

    fireEvent.drop(profilePictureInput, {
        dataTransfer: {
            files: fakeFile,
        },
    });

    const attachmentInput = await screen.findByTestId('attachment-dropzone-input');

    fireEvent.drop(attachmentInput, {
        dataTransfer: {
            files: fakeDoc,
        },
    });

    userEvent.click(await screen.findByText('Uložit'));

    /** TODO Submit was called successfully with image and attachment **/

    await waitFor(() => {
        expect(handleSubmit).toHaveBeenCalled();
        // expect(handleSubmit).toHaveBeenCalledWith({
        //     id: 123,
        //     first_name: 'Test',
        //     last_name: 'Employee',
        //     email: 'test@email.com',
        //     phone: 420123456789,
        //     active: true,
        //     med_exam_date: '2022-02-01',
        //     job_assign_date: '2022-02-12',
        //     working_category: WorkingCategoryEnum.C,
        //     health_limitations: '',
        //     company: null,
        //     attachment: fakeDoc[0],
        //     profile_picture: fakeFile[0],
        // });
    });
});

test('Form method isnt called and errors are shown when inputting wrong values', async () => {
    const handleSubmit = jest.fn();

    render(<EmployeeForm initialValues={new EmployeeDto()} onSubmit={handleSubmit} />);

    const firstNameInput = await screen.findByLabelText('Jméno');
    const phoneInput = await screen.findByLabelText('Telefon');
    const emailInput = await screen.findByLabelText('Email');
    const workCategoryInput = await screen.findByRole('combobox');

    const submitButton = await screen.findByTestId('submit-button');

    userEvent.type(firstNameInput, 'Da'); /** Use too short name **/
    /** Forget last name **/
    userEvent.type(phoneInput, '12346578');
    /** Miss few numbers **/
    userEvent.type(emailInput, 'alfred.dlouhygmail.com'); /** Forget at-sign **/
    /** Forget to select category **/

    await waitFor(async () => {
        expect(firstNameInput).not.toBeNull();
        expect(firstNameInput).toHaveValue('Da');
        expect(await screen.findByTestId('text-input-error')).toBeInTheDocument();
        expect(await screen.findByText('Jméno je příliš krátké')).toBeInTheDocument();

        expect(phoneInput).not.toBeNull();
        expect(phoneInput).toHaveValue('12346578');
        expect(await screen.findByTestId('phone-number-input-error')).toBeInTheDocument();
        expect(await screen.findByText('Číslo musí mít přesně 12 číslic')).toBeInTheDocument();
    });

    // userEvent.click(await screen.findByTestId('category-select-input'));
    // /** https://github.com/testing-library/user-event/issues/189 **/
    // // userEvent.click(document.body);
    // fireEvent.blur(await screen.findByTestId('category-select-input'));

    userEvent.click(submitButton);

    expect(await screen.findByTestId('category-input-error')).toBeInTheDocument();
    expect(await screen.findByText('Kategorie musí být vyplněna')).toBeInTheDocument();

    await waitFor(async () => {
        expect(handleSubmit).not.toHaveBeenCalled();
        expect(await screen.findByTestId('invalid-form-error')).toBeInTheDocument();
        expect(
            await screen.findByText('Ve formuláři jsou chyby. Opravte je a zkuste to prosím znovu.'),
        ).toBeInTheDocument();
    });
});

test('Form submit is called correctly with required inputs', async () => {
    const handleSubmit = jest.fn();

    render(<EmployeeForm initialValues={new EmployeeDto()} onSubmit={handleSubmit} />);

    const firstNameInput = await screen.findByLabelText('Jméno');
    const lastNameInput = await screen.findByLabelText('Příjmení');
    const phoneInput = await screen.findByLabelText('Telefon');
    const emailInput = await screen.findByLabelText('Email');
    const workCategoryInput = await screen.findByRole('combobox');

    const submitButton = await screen.findByTestId('submit-button');

    userEvent.type(firstNameInput, 'Alfred');
    userEvent.type(lastNameInput, 'Dlouhy');
    userEvent.type(phoneInput, '420735445210');
    userEvent.type(emailInput, 'alfred.dlouhy@gmail.com');
    userEvent.selectOptions(workCategoryInput, ['A']);

    expect(firstNameInput).not.toBeNull();
    expect(firstNameInput).toHaveValue('Alfred');

    expect(workCategoryInput).not.toBeNull();
    expect(workCategoryInput).toHaveValue('A');

    userEvent.click(submitButton);

    await waitFor(() => {
        expect(handleSubmit).toHaveBeenCalled();
        expect(handleSubmit.mock.calls[0][0]).toEqual({
            first_name: 'Alfred',
            last_name: 'Dlouhy',
            phone: 420735445210,
            email: 'alfred.dlouhy@gmail.com',
            working_category: 'A',
            active: undefined,
            attachment: undefined,
            company: undefined,
            health_limitations: undefined,
            job_assign_date: undefined,
            med_exam_date: undefined,
            profile_picture: undefined,
        });
        /** Toto pri volani funkce vklada krome SPRAVNEHO objektu vypsaneho nize, taky seznam funkci Formiku,
         *  proto podminka dole neplati a je pouzito reseni vyse **/
        // expect(handleSubmit).toHaveBeenCalledWith({
        //     first_name: 'Alfred',
        //     last_name: 'Dlouhy',
        //     phone: 420735445210,
        //     email: 'alfred.dlouhy@gmail.com',
        //     working_category: 'A',
        //     active: undefined,
        //     attachment: undefined,
        //     company: undefined,
        //     health_limitations: undefined,
        //     job_assign_date: undefined,
        //     med_exam_date: undefined,
        //     profile_picture: undefined,
        // });
    });
});
