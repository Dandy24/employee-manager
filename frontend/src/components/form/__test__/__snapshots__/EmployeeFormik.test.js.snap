// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form renders correctly with proper initial values 1`] = `
<DocumentFragment>
  <form
    action="#"
    data-testid="employee-form-form"
    style="width: 80%; margin-left: 9%;"
  >
    <div
      class="ant-row ant-row-center"
    >
      <div
        data-testid="employee-form"
      >
        <div
          class="ant-row ant-row-center"
        >
          <div
            class="ant-col"
          >
            <span
              class="ant-upload-picture-card-wrapper avatar-uploader profile_picture-empty"
            >
              <div
                class="ant-upload ant-upload-select ant-upload-select-picture-card"
              >
                <span
                  class="ant-upload"
                  role="button"
                  tabindex="0"
                >
                  <input
                    accept=".png,.jpg,.jpeg"
                    data-testid="profile_picture-input"
                    style="display: none;"
                    type="file"
                  />
                  <div>
                    <span
                      aria-label="plus"
                      class="anticon anticon-plus"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="plus"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <defs>
                          <style />
                        </defs>
                        <path
                          d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
                        />
                        <path
                          d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
                        />
                      </svg>
                    </span>
                    <div
                      style="margin-top: 8px;"
                    >
                      Nahrát fotku
                    </div>
                  </div>
                </span>
              </div>
            </span>
          </div>
        </div>
        <div
          style="margin-top: 4.5%;"
        />
        <div
          data-testid="first_name-text-input"
        >
          <label
            data-testid="text-input-label"
            for="first_name"
          >
            Jméno
          </label>
          <input
            class="ant-input"
            data-testid="text-input-field"
            id="first_name"
            name="first_name"
            type="text"
            value=""
          />
        </div>
        <div
          data-testid="last_name-text-input"
        >
          <label
            data-testid="text-input-label"
            for="last_name"
          >
            Příjmení
          </label>
          <input
            class="ant-input"
            data-testid="text-input-field"
            id="last_name"
            name="last_name"
            type="text"
            value=""
          />
        </div>
        <div
          data-testid="phone-number-input"
        >
          <label
            data-testid="phone-number-input-label"
            for="phone"
          >
            Telefon
          </label>
          <div
            class="ant-input-number"
            style="width: 100%;"
          >
            <div
              class="ant-input-number-handler-wrap"
            >
              <span
                aria-disabled="false"
                aria-label="Increase Value"
                class="ant-input-number-handler ant-input-number-handler-up"
                role="button"
                unselectable="on"
              >
                <span
                  aria-label="up"
                  class="anticon anticon-up ant-input-number-handler-up-inner"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="up"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M890.5 755.3L537.9 269.2c-12.8-17.6-39-17.6-51.7 0L133.5 755.3A8 8 0 00140 768h75c5.1 0 9.9-2.5 12.9-6.6L512 369.8l284.1 391.6c3 4.1 7.8 6.6 12.9 6.6h75c6.5 0 10.3-7.4 6.5-12.7z"
                    />
                  </svg>
                </span>
              </span>
              <span
                aria-disabled="false"
                aria-label="Decrease Value"
                class="ant-input-number-handler ant-input-number-handler-down"
                role="button"
                unselectable="on"
              >
                <span
                  aria-label="down"
                  class="anticon anticon-down ant-input-number-handler-down-inner"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="down"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                    />
                  </svg>
                </span>
              </span>
            </div>
            <div
              class="ant-input-number-input-wrap"
            >
              <input
                autocomplete="off"
                class="ant-input-number-input"
                data-testid="phone-number-input-field"
                id="phone"
                label="Telefon"
                name="phone"
                role="spinbutton"
                step="1"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          data-testid="email-text-input"
        >
          <label
            data-testid="text-input-label"
            for="email"
          >
            Email
          </label>
          <input
            class="ant-input"
            data-testid="text-input-field"
            id="email"
            name="email"
            type="text"
            value=""
          />
        </div>
        <div
          data-testid="working_category-category-select"
        >
          <label
            data-testid="category-select-label"
            for="working_category"
          >
            Kategorizace práce
          </label>
          <br />
          <select
            data-testid="category-select-input"
          >
            <option
              data-testid="category-select-option-A"
              value="A"
            >
              A
            </option>
            <option
              data-testid="category-select-option-B"
              value="B"
            >
              B
            </option>
            <option
              data-testid="category-select-option-C"
              value="C"
            >
              C
            </option>
          </select>
        </div>
        <div
          style="padding: 5px 0px; width: 100%;"
        >
          <div
            class="ant-space ant-space-horizontal ant-space-align-center"
            style="width: 100%;"
          >
            <div
              class="ant-space-item"
              style="margin-right: 24px;"
            >
              <label
                for="med_exam_date"
              >
                Datum vstupní prohlídky
              </label>
            </div>
            <div
              class="ant-space-item"
              style="margin-right: 24px;"
            >
              <br />
            </div>
            <div
              class="ant-space-item"
            >
              <div
                class="ant-picker"
                style="width: 100%;"
              >
                <div
                  class="ant-picker-input"
                >
                  <input
                    autocomplete="off"
                    id="med_exam_date"
                    placeholder="Select date"
                    readonly=""
                    size="12"
                    title=""
                    value=""
                  />
                  <span
                    class="ant-picker-suffix"
                  >
                    <span
                      aria-label="calendar"
                      class="anticon anticon-calendar"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="calendar"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
                        />
                      </svg>
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          style="padding: 5px 0px; width: 100%;"
        >
          <div
            class="ant-space ant-space-horizontal ant-space-align-center"
            style="width: 100%;"
          >
            <div
              class="ant-space-item"
              style="margin-right: 24px;"
            >
              <label
                for="job_assign_date"
              >
                Datum nástupu do práce
              </label>
            </div>
            <div
              class="ant-space-item"
              style="margin-right: 24px;"
            >
              <br />
            </div>
            <div
              class="ant-space-item"
            >
              <div
                class="ant-picker"
                style="width: 100%;"
              >
                <div
                  class="ant-picker-input"
                >
                  <input
                    autocomplete="off"
                    id="job_assign_date"
                    placeholder="Select date"
                    readonly=""
                    size="12"
                    title=""
                    value=""
                  />
                  <span
                    class="ant-picker-suffix"
                  >
                    <span
                      aria-label="calendar"
                      class="anticon anticon-calendar"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="calendar"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
                        />
                      </svg>
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          data-testid="health_limitations-text-area"
        >
          <label
            data-testid="text-area-label"
            for="health_limitations"
          >
            Zdravotní omezení
          </label>
          <textarea
            class="ant-input ant-input-lg"
            data-testid="text-area-input"
            name="health_limitations"
            rows="3"
          />
        </div>
        <div
          style="margin-top: 10%;"
        />
        <div
          class="ant-row ant-row-center"
          data-testid="attachments-dropzone"
        >
          <span>
            <div
              class="ant-upload ant-upload-drag"
              style="width: 100%;"
            >
              <span
                class="ant-upload ant-upload-btn"
                role="button"
                tabindex="0"
              >
                <input
                  accept=".pdf"
                  data-testid="attachment-dropzone-input"
                  style="display: none;"
                  type="file"
                />
                <div
                  class="ant-upload-drag-container"
                >
                  <p
                    class="ant-upload-drag-icon"
                  >
                    <span
                      aria-label="inbox"
                      class="anticon anticon-inbox"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="inbox"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="0 0 1024 1024"
                        width="1em"
                      >
                        <path
                          d="M885.2 446.3l-.2-.8-112.2-285.1c-5-16.1-19.9-27.2-36.8-27.2H281.2c-17 0-32.1 11.3-36.9 27.6L139.4 443l-.3.7-.2.8c-1.3 4.9-1.7 9.9-1 14.8-.1 1.6-.2 3.2-.2 4.8V830a60.9 60.9 0 0060.8 60.8h627.2c33.5 0 60.8-27.3 60.9-60.8V464.1c0-1.3 0-2.6-.1-3.7.4-4.9 0-9.6-1.3-14.1zm-295.8-43l-.3 15.7c-.8 44.9-31.8 75.1-77.1 75.1-22.1 0-41.1-7.1-54.8-20.6S436 441.2 435.6 419l-.3-15.7H229.5L309 210h399.2l81.7 193.3H589.4zm-375 76.8h157.3c24.3 57.1 76 90.8 140.4 90.8 33.7 0 65-9.4 90.3-27.2 22.2-15.6 39.5-37.4 50.7-63.6h156.5V814H214.4V480.1z"
                        />
                      </svg>
                    </span>
                  </p>
                  <p
                    class="ant-upload-text"
                  >
                    Klikněte do vyznačené zóny nebo přetáhněte soubor z počítače pro jeho nahrání
                  </p>
                  <p
                    class="ant-upload-hint"
                  >
                    Zde můžete nahrát například pracovní smlouvy, lékařské zprávy atd.
                  </p>
                  <p
                    class="ant-upload-hint"
                  >
                    Pouze pro dokumenty formátu PDF
                  </p>
                </div>
              </span>
            </div>
            <div
              class="ant-upload-list ant-upload-list-text"
            />
          </span>
        </div>
        <div
          class="ant-col ant-col-12 ant-col-offset-9"
          style="margin-top: 3vh;"
        >
          <button
            class="ant-btn ant-btn-primary"
            data-testid="submit-button"
            type="submit"
          >
            <span>
              Uložit
            </span>
          </button>
        </div>
      </div>
    </div>
  </form>
</DocumentFragment>
`;
